app/src/Pages/Portal/Contributions/Utils/ContributionsFields.js

```diff
@@ -366,14 +367,14 @@ export const fields = {
   firstName: {
     label: "Contributor's First Name",
     section: FormSectionEnum.CONTRIBUTOR,
-    component: TextField,
+    component: NameField,
```

Comment:
I would use yup and this would match start with a letter end with a letter and allow single spaces, dashes, and apostrophes and number of chars in the middle (including international chars):
 /^[\p{L}'][ \p{L}'-]*[\p{L}]$/u 

firstName: {
    label: "Contributor's First Name",
    section: FormSectionEnum.CONTRIBUTOR,
    component: TextField,
    validation: Yup.string('Name must only contain name characters')
      .matches(/^[\p{L}'][ \p{L}'-]*[\p{L}]$/u, {
        excludeEmptyString: true,
      })
      .nullable(),
  },